R/aggregate_kicking_stats.R

```diff
@@ -0,0 +1,161 @@
+#' Summarize Kicking Stats
+#'
+#' @description Build columns that aggregate kicking stats at the game level.
+#'
+#' @param pbp A Data frame of NFL play-by-play data typically loaded with
+#' [load_pbp()] or [build_nflfastR_pbp()].
+#' @param weekly If `TRUE`, returns week-by-week stats, otherwise, stats for
+#' the entire Data frame.
+#'
+#' @examples
+#'
+#' \donttest{
+#' pbp <- nflreadr::load_pbp(2021)
+#' calculate_kicking_stats(pbp, weekly = TRUE)
+#' }
+#'
+#' @return a dataframe of kicking stats
+#' @seealso <https://nflreadr.nflverse.com/reference/load_player_stats.html> for the nflreadr function to download this from repo (`stat_type = "kicking"`)
+#' @export
+calculate_kicking_stats <- function(pbp, weekly = FALSE){
+
+  # First, creating a grouping variable object to toggle the weekly argument w/
+  grp_vars <- list("season", "season_type", "week", "team", "player_id")
+  if(!weekly) {
+    grp_vars <- setdiff(grp_vars, "week")
+  }
+  grp_vars <- lapply(grp_vars, as.symbol)
+
+  # Filtering down / creating a base dataset
+  df_fg_or_pat = pbp %>%
+    dplyr::filter(field_goal_attempt == 1 | extra_point_attempt == 1) %>%
+    dplyr::select(
+      game_id,
+      season,
+      week,
+      season_type,
+      team = posteam,
+      player_name = kicker_player_name,
+      player_id = kicker_player_id,
+      dist = kick_distance,
+      field_goal_attempt,
+      fg_res = field_goal_result,
+      extra_point_attempt,
+      pat_res = extra_point_result,
+      fixed_drive,
+      score_differential
+    )
```

Comment:
Can you also make sure it passes devtools::check() locally? I think it'll have some problems at the moment because of the way this filter uses unquoted column names at the moment. 

Brief guide for fixing the naming issue: 

- inside filter/mutate/summarise/group_by, use `.data$field_goal_attempt` and `.data$extra_point_attempt` to refer to piped column names
- inside select statements, use fully quoted character strings, e.g. "game_id", "season", "week", "season_type"

Reply:
Ah, very helpful comment. I would not have caught that! `devtools::check()` runs locally on my machine with no errors, warnings or notes.